examples/systems_of_equations/systems_of_equations_ex8/augment_sparsity_on_contact.C

```diff
@@ -0,0 +1,140 @@
+// local includes
+#include "augment_sparsity_on_contact.h"
+
+// libMesh includes
+#include "libmesh/linear_implicit_system.h"
+
+using namespace libMesh;
+
+AugmentSparsityOnContact::AugmentSparsityOnContact(
+  System& sys)
+  :
+  _sys(sys)
+{}
+
+void AugmentSparsityOnContact::clear_contact_element_map()
+{
+  _contact_element_map.clear();
+}
+
+void AugmentSparsityOnContact::add_contact_element(
+  dof_id_type element_id,
+  dof_id_type other_element_id)
+{
+  std::map<dof_id_type, std::set<dof_id_type> >::iterator it =
```

Comment:
Do we want std::map or LIBMESH_BEST_UNORDERED_MAP here?

std::set is probably a decent choice.